aten/src/ATen/MemoryOverlap.cpp

```diff
@@ -0,0 +1,34 @@
+#include <ATen/MemoryOverlap.h>
+#include <c10/core/Layout.h>
+#include <c10/core/Backend.h>
+
+namespace at {
+
+MemOverlap has_internal_overlap(const Tensor& tensor) {
+  auto* t = tensor.unsafeGetTensorImpl();
+
+  AT_ASSERT(layout_from_backend(tensorTypeIdToBackend(t->type_id())) == kStrided);
```

Comment:
Err, is there really no way to get the layout directly from a tensor?

Reply:
Oh, there's a tensor.layout() function. I should use it